airflow-core/src/airflow/ui/src/utils/query.ts

```diff
@@ -20,6 +20,11 @@ import { useDagServiceGetDagDetails } from "openapi/queries";
 import type { TaskInstanceState } from "openapi/requests/types.gen";
 import { useConfig } from "src/queries/useConfig";
 
+export const getOrderBy = (pre_defined?: string): string[] => {
+  const sortParam = new URLSearchParams(globalThis.location.search).getAll("sort");
+  return (sortParam.length === 0 && pre_defined != undefined) ? [pre_defined] : sortParam;
+};
+
```

Comment:
```suggestion
export const getOrderBy = (defaultOrder?: string): string[] => {
  const sortParam = new URLSearchParams(globalThis.location.search).getAll("sort");
  return (sortParam.length === 0 && defaultOrder !== undefined) ? [defaultOrder] : sortParam;
};

```

We generally don't want to use snake_case in our typescript code. The only ones we do have are generated from the python API.

Reply:
Hello @bbovenzi !

In my last commit I fixed the casing, ran the afore mentioned commands and fixed the things marked in the lint command (that were from my commits).

Anything else I should change?

Also, I noticed that the code failed these tests: https://github.com/apache/airflow/actions/runs/14682302749/job/41221555783. How should I fix it? 
I noticed that it fails where I changed orderBy?: string to orderBy?: string[]. From what I saw, I think it's because it clashes with the definition on the file airflow-core/src/airflow/api_fastapi/core_api/openapi/v1-rest-api-generated.yaml (that I changed but I believe it got overwritten).

Could you confirm this please?

Thank you!

Reply:
`pnpm format` should help solve most of those static checks

Reply:
We may also need to make sure we generate the API spec docs before we run pnpm format

Reply:
Hello @bbovenzi !

I ran pnpm format but the static checks still failed...

When you say API spec docs do you mean this file: airflow-core/src/airflow/api_fastapi/core_api/openapi/v1-rest-api-generated.yaml? And if so how can I generate it? I changed it locally, adapted the code that depended on it and everything seems to be working on my machine. What do I need to do to pass the static checks?

Thank you!

Reply:
Oh I think this is an order of operations with pre-commit

`pre-commit run generate-openapi-spec` and then `pre-commit run ts-compile-format-lint-ui`

Reply:
Hello @bbovenzi !

I had previously ran these operations and noticed that the first one changes the v1-rest-api-generated.yaml file like so:

![image](https://github.com/user-attachments/assets/b9414ebd-bee6-4545-a384-7e3b534c0d45)

This is what is causing the errors in the CI because the Order by argument now has to be an array and it keeps changing to string (which doesn't go along with the changes in the code for the multisort).
Given this, what I meant was if there was a way to define the file and the pre-commit to allow the Order By to be an array? From what I know, it seems like this could fix the CI errors.

Thank you!